docs/application/dotnet/guides/applications/nui-widget-app.md

```diff
@@ -0,0 +1,244 @@
+      ```csharp
+        static void Main(string[] args)
+        {
+            var app = new Program(typeof(MyWidget));
+            app.Run(args);
+        }    
+      ```
+
+   - For using the `NUIWidgetApplication(Dictionary< Type, string > typeInfo)` constructor, in case your widget applications have multiple widget classes.
+
+      For multiple instances, add `<widget-class>` in XML as follow:
+        ```XML
+          <widget-class classid="second" update-period="0">
+            <support-size preview="Widget.png">2x2</support-size>
+          </widget-class>
+        ```
+      You can then modify the code as follows:
+        ```csharp
+        static void Main(string[] args)
+        {
+           Dictionary<System.Type, string> widgetSet = new Dictionary<Type, string>();
+            widgetSet.Add(typeof(MyWidget), "second@org.tizen.example.WidgetTemplate");
+            var app = new Program(widgetSet);
+            app.Run(args);
+        }    
+        ```
+
+4. Define your widget class, which is inherited from [Widget](https://samsung.github.io/TizenFX/latest/api/Tizen.NUI.Widget.html):
+
+   ```csharp
+   class MyWidget : Widget
+   ```
+
+   
+
+5. Override event callback methods of your new class:
+    ```csharp
+    class MyWidget : Tizen.NUI.Widget
+    {
+      protected override void OnCreate(string contentInfo, Window window)
+      {
+        /// Create the UI
+        /// ....
+        base.OnCreate(contentInfo, window);
+      }
+    }
+    ```
+
+   `Widget` class provides the following callback:
+
+    - `OnCreate()` callback is triggered when the widget instance is created.
+
+    - `OnTerminate()` callback is triggered when the widget instance is terminated.
+
+      ```csharp
+      protected override void OnTerminate(string contentInfo, TerminationType type)
+      ```
+
+    - `OnPause()` callback is triggered when the widget instance is paused.
+
+      Your application does the necessary actions when the widget instance becomes invisible because the Application framework can destroy a paused widget instance, for instance, such as the case that memory is lacked.
+
+      ```csharp
+      protected override void OnPause()
+      ```
+
+    - `OnResume()` callback is triggered when the widget instance is resumed.
+
+      ```csharp
+      protected override void OnResume()
+      ```
+
+    - `OnResize()` callback is triggered when the widget instance is resized.
+
+      Your application does the necessary actions to accommodate the new size.
+
+      ```csharp
+      protected override void OnResize(Window window)
+      ```
+
+    - `OnUpdate()` callback is triggered when a widget update event is received.
+
+      Your application does the necessary actions for the widget update. If the `isForce` parameter is true, the widget can be updated even in the pause state.
+      Your application does the necessary actions when the widget instance becomes visible.
```

Comment:
Please add colon instead of period as a code follows:

```suggestion
 Your application does the necessary actions when the widget instance becomes visible:
```